extensions/sd_api_pictures/script.py

```diff
@@ -97,19 +152,51 @@ def input_modifier(string):
 
     global params
 
-    if not params['mode'] == 1:  # if not in immersive/interactive mode, do nothing
-        return string
-
-    if triggers_are_in(string):  # if we're in it, check for trigger words
+    if params['mode'] == 1:  # if not in immersive/interactive mode, do nothing
+            return if_of_is_in(string)
+    if params['mode'] == 2:
+        params['characterfocus'] = False
         toggle_generation(True)
         string = string.lower()
-        if "of" in string:
-            subject = string.split('of', 1)[1]  # subdivide the string once by the first 'of' instance and get what's coming after it
-            string = "Please provide a detailed and vivid description of " + subject
-        else:
-            string = "Please provide a detailed description of your appearance, your surroundings and what you are doing right now"
+        return string
+    if params['mode'] == 0:
+        toggle_generation(False)
+        return string
+
+
+
+# Add NSFW tags if NSFW is enabled, add character sheet tags if character is describing itself
+def create_suffix():
+    global params
+    params['positive_suffix'] = ""
+    params['negative_suffix'] = ""
+
+    # load character data from json, yaml, or yml file
+    if shared.character != 'None':
+        found_file = False
+        folder1 = 'characters'
+        folder2 = 'characters/instruction-following'
+        for folder in [folder1, folder2]:
+            for extension in ["yml", "yaml", "json"]:
+                filepath = Path(f'{folder}/{shared.character}.{extension}')
+                if filepath.exists():
+                    found_file = True
+                    break
+            if found_file:
+                break
+        file_contents = open(filepath, 'r', encoding='utf-8').read()
+        data = json.loads(file_contents) if extension == "json" else yaml.safe_load(file_contents)
+
+    if params['nsfw']:
+        params['positive_suffix'] = params['nsfw_prompts'] + ", " + params['prompt_translation_positive']
+        params['negative_suffix'] = params['anti_nsfw_prompts'] + ", " + params['prompt_translation_negative']
+    if params['characterfocus']:
+        params['positive_suffix'] = data['positive_sd'] + ", " + params['prompt_translation_positive'] if 'positive_sd' in data else params['prompt_translation_positive']
+        params['negative_suffix'] = data['negative_sd'] + ", " + params['prompt_translation_negative'] if 'positive_sd' in data else params['prompt_translation_negative']
```

Comment:
Also, we should probably check for `negative` (not `positive`) in `data` when dealing with negative suffixes